### PR TITLE
feat: Item WooCommerce tab visibility and single-customer order mapping

### PR DESCRIPTION
**Pull request by IMS94**
### 1. Conditional Item WooCommerce Tab Visibility
- Implements conditional visibility for the WooCommerce tab on Items
- Only shows the tab when appropriate WooCommerce Server credentials are configured
- Improves user experience by hiding irrelevant interface elements

### 2. Single-Customer Order Mapping
- Adds `use_single_customer` and `single_customer` fields to WooCommerce Server
- Routes all orders from a WooCommerce server to a single ERPNext Customer
- Implements intelligent address reuse with canonicalization and matching
- Email-based contact matching with phone number backfilling
- Optimized database queries for efficient address/contact lookups
- Comprehensive test suite covering all scenarios
- Maintains full backward compatibility when feature is disabled

## Technical Details
- Address matching uses case-insensitive comparison with proper country code resolution
- Contact reuse strategy prioritizes email as the unique identifier
- Sales Order properly links to chosen addresses and contacts in single-customer mode
- Error handling and fallbacks for edge cases

## Testing
- Added comprehensive test suite for single-customer functionality
- All existing functionality remains unchanged
- Backward compatibility verified

Resolves requirements for simplified customer management in WooCommerce integration scenarios.

### CODE REVIEW

**Comment by IMS94** [comment]
@cursor review

**Comment by IMS94** [comment]
@codex review

**Comment by IMS94** [comment]
@codex review

**Comment by IMS94** [comment]
@cursor review

File: woocommerce_fusion/tasks/sync_sales_orders.py [hunk 1]
```diff
@@ -614,6 +635,204 @@ def create_or_link_customer_and_address(self, wc_order: WooCommerceOrder) -> str
...
+	def canonicalize_address_data(self, raw_data: Dict, address_type: str) -> Dict:
+		"""
+		Canonicalize address data for comparison according to the spec.
+		"""
+		default_country = frappe.get_system_settings("country")
+		# Get country name from country code
+		country_code = raw_data.get("country")
+		if country_code:
+			country = frappe.get_value("Country", {"code": country_code.lower()}) or default_country
+		else:
```
[the remark this answers is not in the document]
Comment (IMS94) [reply woocommerce_fusion/tasks/sync_sales_orders.py:677]: Frappe uses country codes as lowercase

File: woocommerce_fusion/tasks/sync_sales_orders.py [hunk 2]
```diff
@@ -614,6 +635,204 @@ def create_or_link_customer_and_address(self, wc_order: WooCommerceOrder) -> str
...
+				contact = contacts[0]
+				# Found existing contact, optionally backfill phone if missing
+				contact_doc = frappe.get_doc("Contact", contact.name)
+				if phone and not contact_doc.phone and not contact_doc.mobile_no:
+					contact_doc.add_phone(phone, is_primary_mobile_no=1, is_primary_phone=1)
+					contact_doc.save()
+				return contact_doc
+		
+		# No matching contact found, create a new one
+		return self.create_new_contact(raw_data, customer)
```
[the remark this answers is not in the document]
Comment (IMS94) [reply woocommerce_fusion/tasks/sync_sales_orders.py:807]: Still the name is needed. So this is fine.
